File: projectEuler/p4.c
#include<stdio.h>

/**
 * main - Problem 4
 *
 * Description: Find the largest palindrome made from the
 * 	product of two 3-digit numbers.
 *
 * Return: Always 0 (Success)
*/

int main(void)
{
	int a, b, temp, x, y;
	int prod, reverse, max;

	max = 0;

	for (a = 100; a <= 999; ++a)
	{
		for (b = 100; b <= 999; ++b)
		{
			prod = a * b;
			temp = prod;

			reverse = 0;

			while (temp != 0)
			{
				reverse = (reverse * 10) + (temp % 10);
				temp /= 10;
			}

			if (reverse == prod)
			{
				if (prod > max)
				{

					x = a;
					y = b;
					max = prod;
				}
			}
		}
	}

	printf("The largest palindrome made from the product of two 3-digit numbers is %d = %d * %d\n", max, x, y);

	return (0);
}

File: projectEuler/p4.py
#!/usr/bin/env python3
"""
problem 4; Largest palindrome product
"""

max = 0

for a in range(100, 1000):
    for b in range(100, 1000):
        prod = a * b
        temp = prod
        rev = 0

        while temp != 0:
            rev = (rev * 10) + (temp % 10)
            temp //= 10
        if (rev == prod):
            if (prod > max):
                max = prod
                x = a
                y = b

print(f"The largest palindrome made from the product of two 3-digit numbers is {max} = {x} * {y}\n")
